cni/pkg/iptables/iptables_test.go

```diff
@@ -0,0 +1,197 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package iptables
+
+import (
+	"io"
+	"net/netip"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	testutil "istio.io/istio/pilot/test/util"
+	"istio.io/istio/pkg/util/sets"
+)
+
+func TestIptables(t *testing.T) {
+	cases := []struct {
+		name   string
+		config func(cfg *Config)
+	}{
+		{
+			"default",
+			func(cfg *Config) {
+			},
+		},
+	}
+	probeSNATipv4 := netip.MustParseAddr("169.254.7.127")
+	probeSNATipv6 := netip.MustParseAddr("e9ac:1e77:90ca:399f:4d6d:ece2:2f9b:3164")
+	hostProbePorts := sets.New[uint16](8018, 8017, 8016)
+
+	for _, tt := range cases {
+		for _, ipv6 := range []bool{false, true} {
+			t.Run(tt.name+"_"+ipstr(ipv6), func(t *testing.T) {
+				cfg := constructTestConfig()
+				cfg.EnableInboundIPv6 = ipv6
+				tt.config(cfg)
+				ext := &DependenciesStub{}
+				iptConfigurator := NewIptablesConfigurator(cfg, ext, EmptyNlDeps())
+				var probeIP *netip.Addr
+				if ipv6 {
+					probeIP = &probeSNATipv6
+				} else {
+					probeIP = &probeSNATipv4
+				}
+				err := iptConfigurator.CreateInpodRules(hostProbePorts, probeIP)
+				if err != nil {
+					t.Fatal(err)
+				}
+
+				compareToGolden(t, ipv6, tt.name, ext.ExecutedAll)
+			})
+		}
+	}
+}
+
+func TestIptablesHostRules(t *testing.T) {
+	cases := []struct {
+		name   string
+		config func(cfg *Config)
+	}{
+		{
+			"hostprobe",
+			func(cfg *Config) {
+			},
+		},
+	}
+	probeSNATipv4 := netip.MustParseAddr("169.254.7.127")
+	probeSNATipv6 := netip.MustParseAddr("e9ac:1e77:90ca:399f:4d6d:ece2:2f9b:3164")
+
+	for _, tt := range cases {
+		for _, ipv6 := range []bool{false, true} {
+			t.Run(tt.name+"_"+ipstr(ipv6), func(t *testing.T) {
+				cfg := constructTestConfig()
+				cfg.EnableInboundIPv6 = ipv6
+				tt.config(cfg)
+				ext := &DependenciesStub{}
+				iptConfigurator := NewIptablesConfigurator(cfg, ext, EmptyNlDeps())
+				var probeIP *netip.Addr
+				if ipv6 {
+					probeIP = &probeSNATipv6
+				} else {
+					probeIP = &probeSNATipv4
+				}
+				err := iptConfigurator.CreateHostRulesForHealthChecks(probeIP)
```

Comment:
hmm.. I thought a critical component of this proposal was we were not relying on modifying the host networking stack, which varies dramatically with each CNI. Seems that isn't the case.

Does this work for cilium, for example?

Reply:
We are not relying on any redirection rules in the host networking stack, correct.

This was necessary to allow host-level healthchecks thru without relying on host redirection rules, or webhooks (e.g. the current sidecar host probe rewrite stuff).

It does work with cilium and everything else we tested - these rules shouldn't conflict with anything, unless that anything happens to assign special meaning/routing to `169.254.7.127` -> in which case healthchecks would simply fail, and a different special link-local IP could be defined via config.